src/ripple/basics/impl/Log.cpp

```diff
@@ -340,21 +343,58 @@ Logs::format (std::string& output, std::string const& message,
 }
 
 //------------------------------------------------------------------------------
-static std::unique_ptr<beast::Journal> debugJournal_;
 
-beast::Journal const&
-debugJournal()
+class DebugJournal
 {
-    if (!debugJournal_)
-        debugJournal_ = std::make_unique<beast::Journal>();
+private:
+    std::unique_ptr<beast::Journal::Sink> s_;
+    std::mutex m_;
+
+public:
+    DebugJournal () = default;
+
+    DebugJournal (DebugJournal const&) = delete;
+    DebugJournal& operator=(DebugJournal const&) = delete;
+
+    DebugJournal(DebugJournal&&) = delete;
+    DebugJournal& operator=(DebugJournal&&) = delete;
+
+    void
+    set(std::unique_ptr<beast::Journal::Sink> sink)
+    {
+        std::lock_guard<std::mutex> _(m_);
+        s_ = std::move(sink);
+    }
 
-    return *debugJournal_;
+    beast::Journal
+    get()
+    {
+        std::lock_guard<std::mutex> _(m_);
+        if (s_)
+            return beast::Journal (*s_);
+        return {};
+    }
+};
+
+static
+DebugJournal&
+dj()
+{
+    static DebugJournal _;
+    return _;
 }
 
 void
-setDebugJournalSink(beast::Journal::Sink& sink)
+setDebugJournalSink(
+    std::unique_ptr<beast::Journal::Sink> sink)
+{
+    dj().set(std::move(sink));
+}
+
+beast::Journal
+debugJournal()
 {
-    debugJournal_ = std::make_unique<beast::Journal>(sink);
+    return beast::Journal (dj().get());
```

Comment:
This works, but I think it can be simplified to `return dj().get()`.  Just a thought.